### PR TITLE
Change units to make consistent conversion

### PR DESCRIPTION
**Pull request by Chao1009**
### Briefly, what does this PR introduce?
Change default units from ROOT (mm) to GEANT4 (cm) in npdet_to_step

### What kind of change does this PR introduce?
- [x] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No

### Does this PR change default behavior?
It changes the default units in converting the geometry to STEP file.

### CODE REVIEW

**Comment by Chao1009** [comment]
It feels like a bit of hacking to unlock the default units before the conversion.
Unfortunately, the TOCCToStep class (an interface to OpenCascade that does the conversion) does not provide any functionality to change the units.
Also, the OpenCascade interface is not maintained well. So probably a better way is to do ROOT->GDML->STEP.

**Comment by wdconinc** [comment]
> ROOT->GDML->STEP

No, that's the wrong approach. GDML is a meshed format. We will not end up with a CAD STEP file.

**Comment by Chao1009** [comment]
I see, then we may have to continue maintaining this deprecated class.

**Comment by wdconinc** [comment]
> I see, then we may have to continue maintaining this deprecated class.

Yes, it's the only way to get this functionality.

**Comment by veprbl** [comment]
> > ROOT->GDML->STEP
> 
> No, that's the wrong approach. GDML is a meshed format. We will not end up with a CAD STEP file.

I thought, ROOT and GDML can round trip. Is there an incompatible shape?

**Comment by wdconinc** [comment]
ROOT can write to GDML, and yes it's round trip compatible. I was wrong and ROOT to GDML to STEP does not result in a meshed file.

But which GDML to STEP converter are you talking about? Is that the GDML workbench in freeCAD? I don't think that's friendly to batch-mode.

The issue here, changing mm to cm, is going to be easier solved within the code we have than switching the entire workflow.

File: src/geocad/src/TOCCToStep.cxx [hunk 1]
```diff
@@ -68,8 +68,9 @@ void TOCCToStep::OCCDocCreation()
 ////////////////////////////////////////////////////////////////////////////////
 /// Logical fTree creation.
 
-TDF_Label TOCCToStep::OCCShapeCreation(TGeoManager *m)
+TDF_Label TOCCToStep::OCCShapeCreation(TGeoManager *m, double length_unit_to_mm)
 {
+   XCAFDoc_DocumentTool::SetLengthUnit(fDoc, length_unit_to_mm, UnitsMethods_LengthUnit_Millimeter);
```
Comment (Chao1009) [inline src/geocad/src/TOCCToStep.cxx:73]: This single line does the job, others are just interfacing it.

**Comment by Chao1009** [comment]
image
Tested with DIRC (picked points are not exactly the bar ends)

File: src/tools/src/settings.h [hunk 2]
```diff
@@ -23,8 +23,9 @@ struct settings {
   bool            level_set         = false;
   int             global_level      = 1;
   bool            list_all          = false;
-  int     color          = 1;
-  double  alpha          = 1;
+  int             color             = 1;
+  double          alpha             = 1;
+  double          unit_factor       = 10;
```
Comment (wdconinc) [inline src/tools/src/settings.h:28]: For consistency, I'd use the same name, and make sure this isn't a magical constant.
```suggestion
  double          length_unit_to_mm = (TGeoManager::GetDefaultUnits() == TGeoManager::kRootUnits) ? 1 / TGeoUnit::mm : 1 / TGeant4Unit::mm;
```

Comment (wdconinc) [reply src/tools/src/settings.h:28]: And rename `unit_factor` -> `length_unit_to_mm` throughout.

I might even suggest `tgeo_length_unit_in_mm` as a variable name wherever TGeo context is relevant.

Comment (Chao1009) [reply src/tools/src/settings.h]: Yes, including `tgeo` should be more self-explanatory.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**